### DISCUSSION TITLE
Building libfm-qt in Lubuntu environment results in missing options (compress, extract to, extract here) in context menu

### DISCUSSION THREAD

**Discussion by thiyashoksanju** [lxqt/lxqt#2535] [Q&A]
Hi,
  I am trying to build libfm-qt v0.17.1 in Lubuntu 22.04 environment. I was able to compile & build the library successfully. After installing the library below options are gone in context menu inside the file manager
- Compress
![no_compress](https://github.com/lxqt/lxqt/assets/22815534/38f55708-1f9b-4204-9586-d86c310efa33)

- Extract to
- Extract here
![no_extract](https://github.com/lxqt/lxqt/assets/22815534/a00a231f-f264-4c07-981f-8562f9ec6858)

Also "open in terminal" option (F4 - key) not working. It shows "No such file or directory" error.
![Open_in_Terminal](https://github.com/lxqt/lxqt/assets/22815534/72f6ab9e-a2a2-43cb-8e2f-5edc8d4f5592)

 
When i check the Edit -> Preferences -> Advanced settings, "Terminal emulator" & "Archiver integration" drop downs does not have any items.

Note: I did not change any code in v0.17.1. Built the original code as it is. Below are the steps i followed

- Downloaded lxqt-build-tools v 0.11.0, compiled & built using cmake, make & make install. ( Was unable to do make install so i followed "sudo make install"). Meanwhile installed required packages for compilation such as libqt5core5a
- Downloaded libfm-qt v0.17.1 compiled & built using cmake & make (resolved any error regarding dependency packages by installing them). Compile & build was successful.
- Instead of make install i downloaded libfm-qt9_0.17.1-0ubuntu2_amd64.deb and replaced only the libfm-qt.so.9, libfm-qt.so.9.0.0 of this original package by my build library ( i wanted to create a deb package - i extracted original package, replaced only those 2 library files, recreated a new package without any version update or any other thing)
- Now i installed this newly created package using dpkg -i option. Installation success. Restarted the desktop. Now above mentioned options are missing in the file manager.

I don't know whats wrong in my steps. I had a constraint like i have to create an installable .deb package instead of make install. Please guide me through this issue.

Technical info:
OS: Lubuntu 22.04.4 LTS Jammy Jelly Fish
lxqt : 0.17.0

Note: i even checked the arhivers.list & terminal.list inside the package. Nothing is missing. Everything is fine. But the issue happens somewhere which i could not find out.

**Comment by stefonarch** [comment]
The question is: why do you build from source the default version of `libfm-qt` in Lubuntu?

Another question is: why do you use the complete outdated version of LXQt? There is a [ppa to update to 1.4.0 
](https://launchpad.net/~lubuntu-dev/+archive/ubuntu/backports/+packages).

**Reply by thiyashoksanju** [reply]
I am working on customizing the specified version for a learning project. I used Lubuntu 22.04 where lxqt v0.17.0 is the default DE. While doing the customization i found this issue. Just confirmed that this issue is not because of the code change. Even if i build the source code without any change, that build also causing the issue

**Reply by thiyashoksanju** [reply]
Additional info.

The size of library from original package : 1.4MB
The size of library built from original source in my environment: 1.8MB (without a single code change)

**Reply by stefonarch** [reply]
Lubuntu usually patches many packages - we can't know what they did.  You  could try to compile pcmanfm-qt against your libfm-qt.
